lab1/src/main/java/App/App.java

```diff
@@ -0,0 +1,73 @@
+package App;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *
+ * @author bratizgut
+ */
+public class App {
+
+    private final Receiver reciever;
+    private final Sender sender;
+
+    private final ConnectionStatus status;
+
+    private Thread mainThread = Thread.currentThread();
+    private boolean isRunning = true;
+
+    public App(String IPaddress, int port) throws IOException {
+        status = new ConnectionStatus(5000);
+        reciever = new Receiver(status, IPaddress, port);
+        sender = new Sender(IPaddress);
+
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                isRunning = false;
+                try {
+                    mainThread.join();
```

Comment:
Расскажите почему именно join, как работает join у треда и какие кейсы покроет это решение.

Reply:
@Aloomenon join использую для того, чтобы ждать завершения работы потока, в котором происходит основная работа программы(в данном случае MainThread). Прочитал, что таким образом можно поддерживать работу JVM(When all the hooks have finished it will then run all uninvoked finalizers if finalization-on-exit has been enabled. Finally, the virtual machine will halt.) для того, чтобы гарантировать закрытие сокета.